nvflare/app_common/workflows/base_fedavg.py

```diff
@@ -0,0 +1,176 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+from abc import abstractmethod
+from typing import List
+
+from nvflare.apis.fl_constant import FLMetaKey
+from nvflare.app_common.abstract.fl_model import FLModel
+from nvflare.app_common.aggregators.weighted_aggregation_helper import WeightedAggregationHelper
+from nvflare.app_common.app_constant import AppConstants
+from nvflare.app_common.app_event_type import AppEventType
+from nvflare.app_common.utils.fl_model_utils import FLModelUtils
+from nvflare.security.logging import secure_format_exception
+
+from .model_controller import ModelController
+
+
+class FedAvgModelControllerSpec(ModelController):
```

Comment:
Why do we need this class?

Can we just have `BaseFedAvg` extends `ModelController` ?

Reply:
Yes, good point. The Spec class is not needed. I'm deriving BaseFedAvg directly from ModelController now.